pkgs/applications/networking/mailreaders/mblaze/default.nix

```diff
@@ -26,6 +29,16 @@ stdenv.mkDerivation rec {
     install -Dm644 -t $out/share/zsh/site-functions contrib/_mblaze
   '' + lib.optionalString (ruby != null) ''
     install -Dt $out/bin contrib/msuck contrib/mblow
+
+    for x in `find $out/bin "(" -type f -executable ")" -or -type l`
+    do
+      wrapProgram $x \
+        --prefix PATH : $out/bin \
+        --prefix PATH : ${lib.makeBinPath [
+	  coreutils file gawk gnugrep gnused less nano
+	]} \
+        --set EDITOR nano
```

Comment:
This will override the user’s editor choice, which is an impurity we should definitely allow. The only reason I suggested it for testing is that your normal EDITOR would probably not be a full path, and we were emptying PATH. Similarly, we probably shouldn’t include less in the wrapper, because a user might configure a different PAGER. Overall, it comes down to whether something is an implementation detail (like awk) or whether something is a property of the environment the program is running in that should be customisable (like the editor or pager). Does that make sense?